app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt

```diff
@@ -47,29 +68,50 @@ class WireNotificationManager @Inject constructor(
         messagesManager.handleNotification(notificationsList, userId)
     }
 
-    // TODO: to be changed as soon as we get the qualifiedID from the notification payload
     /**
+     * Showing notifications for some other user then Current one requires to change user on opening Notification,
+     * which is not implemented yet.
+     * So for now we show notifications only for Current User
+     *
      * return the userId if the user is authenticated and null otherwise
      */
-    @Suppress("NestedBlockDepth")
     private fun checkIfUserIsAuthenticated(userId: String): QualifiedID? =
-        coreLogic.getAuthenticationScope().getSessions().let {
-            when (it) {
-                is GetAllSessionsResult.Success -> {
-                    for (sessions in it.sessions) {
-                        if (sessions.userId.value == userId)
-                            return@let sessions.userId
-                    }
-                    null
+        when (val currentSession = coreLogic.getAuthenticationScope().session.currentSession()) {
+            is CurrentSessionResult.Success -> currentSession.authSession.userId
+            else -> null
+        }
+
+    /**
+     * Infinitely listen for the new IncomingCalls, notify about it and do additional actions if needed.
+     * Can be used for listening for the Notifications when the app is running.
+     * @param userIdFlow Flow of QualifiedID of User
+     * @param isAppVisibleFlow StateFlow that informs if app is currently visible,
+     * so we can decide: should we show notification, or just open the IncomingCall screen
+     */
+    suspend fun observeIncomingCalls(
+        isAppVisibleFlow: StateFlow<Boolean>,
```

Comment:
AFAIK the Android system will handle this since you added `setFullScreenIntent` with a pending intent in the notification builder

Reply:
not really. From the docs: 
> If the user's device is unlocked, the notification appears in an expanded form that includes options for handling or dismissing the notification.

so if the device is unlocked (no matter if our app is displayed or not) notification with a FullScreenIntent will be displayed as a notification, no opening the Intent.

Reply:
Yes if the device is unlocked --> a notification will be displayed
if it is locked --> a full screen be displayed